packages/smooth_app/lib/pages/personalized_ranking_page.dart

```diff
@@ -58,9 +60,41 @@ class _PersonalizedRankingPageState extends State<PersonalizedRankingPage>
         context.watch<ProductPreferences>();
     context.watch<LocalDatabase>();
     final AppLocalizations appLocalizations = AppLocalizations.of(context);
+
+    Future<void> _addToList() async {
+      final LocalDatabase localDatabase = context.read<LocalDatabase>();
+      final DaoProductList daoProductList = DaoProductList(localDatabase);
+      await ProductListUserDialogHelper(daoProductList)
+          .showBulkInsertUserListsDialog(
+        context,
+        widget.barcodes,
+      );
+    }
+
+    void handlePopUpClick(String value) {
+      switch (value) {
+        case 'add_to_list':
+          _addToList();
+          break;
+      }
+    }
```

Comment:
```suggestion
    Future<void> handlePopUpClick(String value) async {
      switch (value) {
        case 'add_to_list':
          await _addToList();
          break;
      }
      throw Exception('Unknown case $value');
    }
```

Besides, I think we would be better off without this method and putting the code directly inline in `onSelected`, but that's another story.